x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -134,6 +134,70 @@ static DeprecationIssue discoveryConfigurationCheck(List<NodeInfo> nodeInfos, Li
         return null;
     }
 
+    static DeprecationIssue watcherNotificationsSecureSettingsCheck(List<NodeInfo> nodeInfos, List<NodeStats> nodeStats) {
+        List<String> nodesFound = nodeInfos.stream()
+                .filter(nodeInfo -> false == nodeInfo.getSettings().getByPrefix("xpack.notification.email.account.")
+                        .filter(s -> s.endsWith(".smtp.password")).isEmpty())
+                .map(nodeInfo -> nodeInfo.getNode().getName())
+                .collect(Collectors.toList());
+        if (nodesFound.size() > 0) {
+            return new DeprecationIssue(DeprecationIssue.Level.CRITICAL,
```

Comment:
I'm concerned that, if there's both email and hipchat accounts set up insecurely, this will produce a deprecation warning for only the email account until that's resolved, then switch to warning about the hipchat account.  I think we should either break this out into many separate checks, or merge the nodesFound lists before creating the `DeprecationIssue`.  

My initial preference is for the second one, with deprecation text something like:
```
new DeprecationIssue(DeprecationIssue.Level.CRITICAL,
                    "Watcher notification accounts' authentication settings must be defined securely",
                    "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-7.0.html" +
                        "#watcher-notifications-account-settings",
                    "nodes which have insecure notification account settings are: " + nodesFound);
```

Although, if you feel strongly that we should have a bunch of separate checks so we can't specify the type of account for each warning, I would be okay with that too.